packages/language-server/src/plugins/typescript/features/DocumentSymbolsProvider.ts

```diff
@@ -33,7 +34,8 @@ export class DocumentSymbolsProviderImpl implements DocumentSymbolsProvider {
 					Range.create(
 						document.positionAt(document.astroMeta.frontmatter.startOffset as number),
 						document.positionAt(document.astroMeta.frontmatter.endOffset as number)
-					)
+					),
+					document.getURL()
```

Comment:
Previously the symbols for the namespace didn't have the proper URI assigned to them, which meant that in certain case you wouldn't navigate to the proper files when trying to go a document's frontmatter or template